source/llassetgen/source/DistanceTransform.cpp

```diff
@@ -219,6 +219,41 @@ namespace llassetgen {
                 if (inputAt({x, y})) outputAt({x, y}) *= -1;
     }
 
+
+
+    template<bool fill>
+    void ParabolaEnvelope::edgeDetection(DimensionType offset, DimensionType pitch, DimensionType length) {
+        InputType prev = 0;
+        DimensionType begin = 0;
+        for(DimensionType j = 0; j < length; ++j) {
+            InputType next = inputAt(offset+j*pitch);
+            if(next == prev)
+                continue;
+            DimensionType end = (next) ? j : j-1;
+            if(fill)
+                for(DimensionType i = begin; i < end; ++i) {
+                    if(begin == 0 && !prev)
+                        outputAt(offset+i*pitch) = square(end-i);
+                    else
+                        outputAt(offset+i*pitch) = square((i < (end+begin)/2) ? i-begin+1 : end-i);
+                }
+            prev = next;
+            begin = end+1;
+            outputAt(offset+end*pitch) = 0;
+        }
+        if(fill)
+            for(DimensionType i = begin; i < length; ++i) {
+                if(begin == 0 && !prev)
+                    outputAt(offset+i*pitch) = 1E20F;
```

Comment:
Is this some kind of infinity? `std::numeric_limits<float>::infinity()`

Reply:
Either way, it is a magic number and should be referenced through an own constant or symbol.

Reply:
std::numeric_limits<OutputType>::infinity() does not work here, but I'll use std::numeric_limits<OutputType>::max() instead